app/javascript/components/redfish-server-firmware-update-dialog.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import FirmwareUpdate from "./forms/firmware-update"
+import { handleApiErrorHooks, fetchFirmwareBinariesForServer, createFirmwareUpdateRequest } from "../utils/api";
+import { selectedPhysicalServers } from "../utils/common";
+
+const RedfishServerFirmwareUpdateDialog = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [firmwareBinaries, setFirmwareBinaries] = useState([]);
+  const [error, setError] = useState('');
+  const [values, setValues] = useState({});
+  const physicalServerIds = selectedPhysicalServers();
```

Comment:
This assignment will be invoked on every component render. Is that intentional?

Reply:
Yes, everytime a popup is opened it should get invoked. I didn't want to use state variable for it because it would complicates things; namely because I can't set state variable inside useEffect and then use it immediately... For example, this won't work :

```javascript
const [ids, setIds] = useState([]);

useEffect(() => {
  setIds([1, 2, 3])
  console.log(ids);  // <---- this will print empty list because state was NOT updated immediately by previous line
});
```

If I wanted to implement it as a state variable then I would need to add another useEffect(..., [ids]) to use the calculated value. If you know what I mean